rules/library.bzl

```diff
@@ -280,34 +336,6 @@ def _xcframework_slice(*, xcframework_name, identifier, platform, platform_varia
             xcframework_name,
         ))
 
-    # Collecting interface files for VFS
-    import_headers = native.glob(
-        ["%s/**/*.h" % path],
-    )
-    import_module_maps = native.glob(
-        ["%s/**/*.modulemap" % path],
-        allow_empty = True,
-    )
-    import_swiftmodules = native.glob(
-        ["%s/**/*.swiftmodule/*.*" % path],
-        allow_empty = True,
-    )
-    if len(import_module_maps) > 0:
-        import_module_map = import_module_maps[0]
-    else:
-        import_module_map = None
-
-    imported_framework_name = _find_imported_framework_name(import_headers)
-    vfs_framework_name = imported_framework_name if imported_framework_name else xcframework_name
-    framework_vfs_overlay(
-        name = resolved_target_name_vfs_overlay,
-        framework_name = vfs_framework_name,
-        modulemap = import_module_map,
-        swiftmodules = import_swiftmodules,
-        hdrs = import_headers,
-        tags = _MANUAL,
-        extra_search_paths = xcframework_name,
-    )
```

Comment:
Moving this logic to the individual branches, there are some differences between each type of framework and it made more sense to do that in each individual branch.